### PR TITLE
xfdashboard: Maybe a Gnome shell and macOS Expose like dashboard for Xfce

### PR DESCRIPTION
**Pull request by tbeckerson**
New Module.

xfdashboard provides a GNOME shell dashboard and macOS Mission Control (Exposé and Spaces) like interface for use with the Xfce desktop.
https://docs.xfce.org/apps/xfdashboard/start

### CODE REVIEW

**Comment by lunar-linux-b01** [comment]
Can one of the admins verify this patch?

**Comment by florin65** [comment]
There are 2 approaches for the depends and optional_depends:
   1. is to add all the depends in the DEPENDS file (which needs to use ldd on the binaries to see what libs are used at runtime)
  2. found the main depends (and do not add their depends because it have to be already installed before). Some careful have to be used on the optional depends of the depends, possible circular depends, etc.

Useful is to study the other distros (Arch, Gentoo, LFS) module and see what depends are used for build there. This is what offers you some work already done  (and tested) for those distros. 
Please see for example https://aur.archlinux.org/packages/xfdashboard

From the configure file:
------------------------------
Some depends have to be added. One is clutter (it depends on cogl). 
If you are looking into the configure file these are the optional depends (line 45858)
  xcomposite
  xdamage
  xinerama

Try to build it by hand first, use the new created/proposed module after and in the end try to add the minimal/mandatory depends and see most/all of the optional depends needed for the final module.

From the compile log (in lunar):
----------------------------
Library versions used:
  libwnck:          3.36.0
  clutter:          1.26.4
  cogl:             1.26.4
  gtk+:             3.24.18
  glib:             2.64.2
  gio:              2.64.2
  gio-unix:         2.64.2
  garcon:           0.6.4
  libxfce4ui:       4.14.1
  libxfce4util:     4.14.0
  libxconf:         4.14.1

Tools used:
  glib-genmarshal:  /usr/bin/glib-genmarshal
  glib-mkenums:     /usr/bin/glib-mkenums

Features enabled:
  Backends:         x11 gdk
  XComposite:       yes
  XDamage:          yes
  Xinerama:         yes

So, in conclusion, for this module you can probably need a DEPENDS file like this one instead:

depends - clutter, garcon, libwnck3
optional depends  - libXinerama, libXdamage, libXcomposite

with the associated params (see the configure file).

This might be the first version of the DEPENDS file.
Some more checks have to be done on the optional_depends (to see if are depends for the new depends and avoid possible circular depends).

I hope to be useful all these considerations to you.

**Comment by lazyape** [comment]
@florin65 @tbeckerson
i try the module in a system without clutter,xfce but with gnome installed.
selecting y to gobject-introspection and gtk-doc in all the questions.

what i see is that `lin xfdashboard` will try to install the following (they are hard dependencies to xfce4-panel)
xfce4-panel
exo
libxfce4ui
libxfce4util
xfce4-icon-theme
garcon
libwnck3

i get an error cause clutter is not present. 
so for me the only change needed here is the addition of

+ depends clutter

i really don't understand the necessity of adding the entire list of the hardware dependencies.
for me simplicity is better than complexity on this. Lunar has the appropriate tools to overcome
the problem if there is a deletion {on purpose or not} of a library from a package dependency.

**Comment by tbeckerson** [comment]
@florin65 

Thank you very much for the guidance! I very much appreciate it.

- Leaving out clutter was a definite mistake on my part, which I will correct in a new commit.

- The optional_depends - libXinerama, libXdamage, libXcomposite: They're all implicit features, and libxdamage and libxcomposite are installed by cogl, which is a hard dependency of clutter. But I'll include an optional_depends for libXinerama for certain.

- As @lazyape mentioned above, the other dependencies are installed by xfce4-panel or _it's_ dependencies. I left those out intentionally. If you want them included, I'll defer to your judgement.

**Comment by florin65** [comment]
I'm not sure is xfce4-panel is mandatory or not. It's in the DEPENDS file now. It should try an install with it commented and xfce4-panel unistalled.

**Comment by tbeckerson** [comment]
DEPENDS are updated:

@florin65 is correct, xfdashboard does not require xfce4-panel.

I've added libwnck3, clutter, garcon, and libxfce4ui as depends, and libXinerama as an optional. Everything else required is pulled in as part of the dependency chain.

Thanks again for the detailed attention to this!
